core/src/main/scala/com/pingcap/tispark/TiBatchWrite.scala

```diff
@@ -428,6 +460,39 @@ class TiBatchWrite(@transient val df: DataFrame,
     }
   }
 
+  private def generateDataToBeRemovedRdd(rdd: RDD[WrappedRow], startTs: TiTimestamp) = {
+    rdd
```

Comment:
Do we need the consider the duplicated remove case?(a row is removed more than one times as it conflicts with more than one input rows)

Reply:
Data to be inserted will be deduplicated first if the table has pk or unique index.

Reply:
is there any test cases?

Reply:
Yes. tests defined in AddingIndexReplaceSuite.